targets/psa_switch/pswitch_CLI.py

```diff
@@ -37,41 +50,69 @@ def __init__(self, pre_type, standard_client, mc_client, pswitch_client):
                                         standard_client, mc_client)
         self.pswitch_client = pswitch_client
 
+    @handle_bad_input
     def do_set_queue_depth(self, line):
         "Set depth of one / all egress queue(s): set_queue_depth <nb_pkts> [<egress_port>]"
         args = line.split()
-        depth = int(args[0])
+        depth = self.parse_int(args[0])
         if len(args) > 1:
-            port = int(args[1])
+            port = self.parse_int(args[1])
             self.pswitch_client.set_egress_queue_depth(port, depth)
         else:
             self.pswitch_client.set_all_egress_queue_depths(depth)
 
+    @handle_bad_input
     def do_set_queue_rate(self, line):
         "Set rate of one / all egress queue(s): set_queue_rate <rate_pps> [<egress_port>]"
         args = line.split()
-        rate = int(args[0])
+        rate = self.parse_int(args[0])
         if len(args) > 1:
-            port = int(args[1])
+            port = self.parse_int(args[1])
             self.pswitch_client.set_egress_queue_rate(port, rate)
         else:
             self.pswitch_client.set_all_egress_queue_rates(rate)
 
+    @handle_bad_input
     def do_mirroring_add(self, line):
         "Add mirroring mapping: mirroring_add <mirror_id> <egress_port>"
         args = line.split()
-        mirror_id, egress_port = int(args[0]), int(args[1])
-        self.pswitch_client.mirroring_mapping_add(mirror_id, egress_port)
-
+        mirror_id = self.parse_int(args[0], "mirror_id")
+        egress_port = self.parse_int(args[1], "egress_port")
+        config = MirroringSessionConfig(port=egress_port)
+        self.pswitch_client.mirroring_session_add(mirror_id, config)
+
+    @handle_bad_input
+    def do_mirroring_add_mc(self, line):
+        "Add mirroring session to multicast group: mirroring_add_mc <mirror_id> <mgrp>"
+        args = line.split()
+        self.exactly_n_args(args, 2)
+        mirror_id = self.parse_int(args[0], "mirror_id")
+        mgrp = self.parse_int(args[1], "mgrp")
+        config = MirroringSessionConfig(mgid=mgrp)
+        self.pswitch_client.mirroring_session_add(mirror_id, config)
+        print "Associating multicast group", mgrp, "to mirroring session", mirror_id
+
+    @handle_bad_input
     def do_mirroring_delete(self, line):
         "Delete mirroring mapping: mirroring_delete <mirror_id>"
-        mirror_id = int(line)
+        mirror_id = self.parse_int(line)
```

Comment:
`self.parse_int(line, "mirror_id")`

Reply:
thanks for catching - fixing this in next commit